// discord.js-commando/src/commands/util/eval.js
const util = require('util');
const discord = require('discord.js');
const tags = require('common-tags');
const escapeRegex = require('escape-string-regexp');
const { Command } = require('discord.js-commando');
const Discord = require('discord.js')
const tokenuyari = `--Bu bilgi gizlidir.--`
const nl = '!!NL!!';
const nlPattern = new RegExp(nl, 'g');

module.exports = class EvalCommand extends Command {
    constructor(client) {
        super(client, {
            name: 'eval',
            group: 'util',
            memberName: 'eval',
            description: 'Yazılan kodu çalıştırır.',
            details: 'Bot Sahibinin Komudu.',
            ownerOnly: true,

            args: [
                {
                    key: 'script',
                    prompt: 'Hangi kodu çalıştırmak istersin?',
                    type: 'string'
                }
            ]
        });

        this.lastResult = null;
    }

async run (message, args)  {
          const msg = message;
        const client = msg.client;
        const objects = client.registry.evalObjects;
        const lastResult = this.lastResult;
  args = args;
    if(!args) {
        const embed = new Discord.RichEmbed()
            .setDescription(`\`${msg.guild.commandPrefix}yardım ${message.content.split(" ")[0].slice(msg.guild.commandPrefix.length)}\` yazarak doğru kullanıma bakabilirsin!`)
            .setColor('RANDOM')
            .setTimestamp()
        message.channel.send({embed})
        return
    }
    const code = args.script;
    if(code.match(/(client.token)/g)) {
        const newEmbed = new Discord.RichEmbed()
            .addField('Hata çıktı;', `\`\`\`xl\n${tokenuyari}\`\`\``)
            .setColor('RANDOM');
        message.channel.send(newEmbed);
        return
    }

    function clean(text) {
        if (typeof text !== 'string')
            text = require('util').inspect(text, { depth: 0 })
        text = text
            .replace(/`/g, '`' + String.fromCharCode(8203))
            .replace(/@/g, '@' + String.fromCharCode(8203))
        return text;
    };

    const evalEmbed = new Discord.RichEmbed().setColor('RANDOM')
    try {
        var evaled = clean(await eval(code));
        if(evaled.startsWith('NDc4O')) evaled = tokenuyari;
        if (evaled.constructor.name === 'Promise') evalEmbed.setDescription(`\`\`\`\n${evaled}\n\`\`\``)
        else evalEmbed.setDescription(`\`\`\`xl\n${evaled}\n\`\`\``)
        const newEmbed = new Discord.RichEmbed()
            .addField('📥 Giriş', `\`\`\`javascript\n${code}\n\`\`\``)
            .addField('📤 Çıkış', `\`\`\`xl\n${evaled}\`\`\``)
            .setColor('RANDOM')
        message.channel.send(newEmbed);
    }
    catch (err) {
        evalEmbed.addField('Hata çıktı;', `\`\`\`xl\n${err}\n\`\`\``);
        evalEmbed.setColor('RANDOM');
        message.channel.send(evalEmbed);
    }
}
}
