// discord.js-commando/src/commands/commands/disable.js
const { oneLine } = require('common-tags');
const Command = require('../base');

module.exports = class DisableCommandCommand extends Command {
	constructor(client) {
		super(client, {
			name: 'disable',
			aliases: ['disable-command', 'cmd-off', 'command-off' , 'komutu-kapat'],
			group: 'commands',
			memberName: 'disable',
			description: 'Bir komutu veya komut grubunu devre dışı bırakır.',
			details: oneLine`
				Bir komut veya komut grubu adını yazarak devre dışı bıraka bilirsiinz sadece yöneticiler kullanabilir.
			`,
			examples: ['disable util', 'disable Utility', 'disable prefix'],
			guarded: true,

			args: [
				{
					key: 'cmdOrGrp',
					label: 'command/group',
					prompt: 'Hangi komutu veya grubu devre dışı bırakmak istersiniz?',
					type: 'group|command'
				}
			]
		});
	}

	hasPermission(msg) {
		if(!msg.guild) return this.client.isOwner(msg.author);
		return msg.member.hasPermission('ADMINISTRATOR') || this.client.isOwner(msg.author);
	}

	run(msg, args) {
		if(!args.cmdOrGrp.isEnabledIn(msg.guild, true)) {
			return msg.reply(
				`\`${args.cmdOrGrp.name}\` ${args.cmdOrGrp.group ? 'komutu' : 'grubu'} zaten devre dılı.`
			);
		}
		if(args.cmdOrGrp.guarded) {
			return msg.reply(
				`Devre dışı bırakamassınız \`${args.cmdOrGrp.name}\` ${args.cmdOrGrp.group ? 'komutu' : 'grubu'}.`
			);
		}
		args.cmdOrGrp.setEnabledIn(msg.guild, false);
		return msg.reply(`Devre dışı bırakıldı \`${args.cmdOrGrp.name}\` ${args.cmdOrGrp.group ? 'komutu' : 'grubu'}.`);
	}
};
