// discord.js-commando/src/commands/commands/enable.js
const { oneLine } = require('common-tags');
const Command = require('../base');

module.exports = class EnableCommandCommand extends Command {
	constructor(client) {
		super(client, {
			name: 'enable',
			aliases: ['enable-command', 'cmd-on', 'command-on' , 'aktifet'],
			group: 'commands',
			memberName: 'enable',
			description: 'Devre dışı bırakılmış komut veya grupları aktif eder.',
			details: oneLine`
				Devre dışı bırakılmış komut veya grupları aktif eder bu komutu sadece yöneticiler kullanabilir.
			`,
			examples: ['enable util', 'enable Utility', 'enable prefix'],
			guarded: true,

			args: [
				{
					key: 'cmdOrGrp',
					label: 'command/group',
					prompt: 'Hangi komut veya grubu aktif etmek istersin?',
					type: 'group|command'
				}
			]
		});
	}

	hasPermission(msg) {
		if(!msg.guild) return this.client.isOwner(msg.author);
		return msg.member.hasPermission('ADMINISTRATOR') || this.client.isOwner(msg.author);
	}

	run(msg, args) {
		const group = args.cmdOrGrp.group;
		if(args.cmdOrGrp.isEnabledIn(msg.guild, true)) {
			return msg.reply(
				`\`${args.cmdOrGrp.name}\` ${args.cmdOrGrp.group ? 'komutu' : 'grubu'} zaten etkin${
					group && !group.enabled ? `, fakat \`${group.name}\` grup devre dışı, bu yüzden hala kullanılamaz` : ''
				}.`
			);
		}
		args.cmdOrGrp.setEnabledIn(msg.guild, true);
		return msg.reply(
			`Aktif edildi \`${args.cmdOrGrp.name}\` ${group ? 'komutu' : 'grubu'}${
				group && !group.enabled ? `, fakat \`${group.name}\` grubu devre dışı, bu yüzden hala kullanılamaz` : ''
			}.`
		);
	}
};
