// discord.js-commando/src/commands/commands/reload.js
const { oneLine } = require('common-tags');
const Command = require('../base');

module.exports = class ReloadCommandCommand extends Command {
	constructor(client) {
		super(client, {
			name: 'reload',
			aliases: ['reload-command','yenidenyükle'],
			group: 'commands',
			memberName: 'reload',
			description: 'Grubu veya komutu yeniden yükler.',
			details: oneLine`
			Komut adı veya grub adını girerek yeniden yükler sadece bot sahibi kullana bilir.
			`,
			examples: ['reload some-command'],
			ownerOnly: true,
			guarded: true,

			args: [
				{
					key: 'cmdOrGrp',
					label: 'command/group',
					prompt: 'Hangi komutu veya grubu yeniden başlatmak istersin?',
					type: 'group|command'
				}
			]
		});
	}

	async run(msg, args) {
		const { cmdOrGrp } = args;
		const isCmd = Boolean(cmdOrGrp.groupID);
		cmdOrGrp.reload();

		if(this.client.shard) {
			try {
				await this.client.shard.broadcastEval(`
					if(this.shard.id !== ${this.client.shard.id}) {
						this.registry.${isCmd ? 'komutlar' : 'gruplar'}.get('${isCmd ? cmdOrGrp.name : cmdOrGrp.id}').reload();
					}
				`);
			} catch(err) {
				this.client.emit('warn', `Diğer parçalara yüklenirken hata oluştu.`);
				this.client.emit('error', err);
				if(isCmd) {
					await msg.reply(`Yeniden yüklendi \`${cmdOrGrp.name}\` komutu, ancak diğer parçalara yeniden yüklenemedi.`);
				} else {
					await msg.reply(
						`\`${cmdOrGrp.name}\` Grubu yeniden yüklendi ancak diğer parçalara yeniden yüklenemedi.`
					);
				}
				return null;
			}
		}

		if(isCmd) {
			await msg.reply(`Yeniden yüklendi \`${cmdOrGrp.name}\` komutu${this.client.shard ? ' tüm parçalara' : ''}.`);
		} else {
			await msg.reply(
				`\`${cmdOrGrp.name}\` group${this.client.shard ? ' Grubu yeniden yüklendi tüm parçalarda aktif' : ''}.`
			);
		}
		return null;
	}
};
