// discord.js-commando/src/commands/commands/unload.js
const { oneLine } = require('common-tags');
const Command = require('../base');

module.exports = class UnloadCommandCommand extends Command {
	constructor(client) {
		super(client, {
			name: 'unload',
			aliases: ['unload-command' , 'komutuboşlat'],
			group: 'commands',
			memberName: 'unload',
			description: 'İstediğiniz bir komutu boşlatır.',
			details: oneLine`
				
Argüman, bir komutun adı / kimliği (kısmi veya tamamı) olmalıdır.
Bu komutu yalnızca bot sahibi (sahipleri) kullanabilir.
			`,
			examples: ['unload some-command'],
			ownerOnly: true,
			guarded: true,

			args: [
				{
					key: 'command',
					prompt: 'Hangi komutu boşlatmak istersiniz?',
					type: 'command'
				}
			]
		});
	}

	async run(msg, args) {
		args.command.unload();

		if(this.client.shard) {
			try {
				await this.client.shard.broadcastEval(`
					if(this.shard.id !== ${this.client.shard.id}) this.registry.commands.get('${args.command.name}').unload();
				`);
			} catch(err) {
				this.client.emit('warn', `Komutu diğer parçalara yüklenirken hata oluştu`);
				this.client.emit('error', err);
				await msg.reply(`Boşaltılan komut \`${args.command.name}\` ama diğer parçalara boşaltılamadı.`);
				return null;
			}
		}

		await msg.reply(`Boşaltılan komut \`${args.command.name}\` ${this.client.shard ? ' tüm parçalarda boşlatıldı' : ''}.`);
		return null;
	}
};
